File: Backend/src/main/java/vn/com/vti_crm/service/UserService.java
package vn.com.vti_crm.service;

import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Service;

import vn.com.vti_crm.domain.User;

@Service
public interface UserService {

	public User createUser(User user);

	public Optional<User> getUserById(int id);

	public Optional<User> updateUserTeam(int teamId, int userId);

	public Optional<User> removeUserTeam(int teamId, int userId);

	public List<User> getAllUsers();

	public void deleteUser(User user);

	public User updateUser(User user);

}

File: Frontend/src/app/service/user/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  private baseUserUrl = 'http://localhost:8080/admin/users';

  constructor(private http: HttpClient) {}

  createUser(User: Object): Observable<Object> {
    return this.http.post(`${this.baseUserUrl}`, User);
  }

  getUser(id: number): Observable<any> {
    return this.http.get(`${this.baseUserUrl}/${id}`);
  }

  getUsersList(): Observable<any> {
    return this.http.get(`${this.baseUserUrl}`);
  }

  updateUser(id: number, value: any): Observable<Object> {
    return this.http.put(`${this.baseUserUrl}/${id}`, value);
  }

  deleteUser(id: number): Observable<any> {
    return this.http.delete(`${this.baseUserUrl}/${id}`, {
      responseType: 'text',
    });
  }

  addUserTeam(userId: number, teamId: number): Observable<Object> {
    return this.http.post(`${this.baseUserUrl}/${userId}/join-team/${teamId}`, {
      responseType: 'json',
    });
  }

  removeUserTeam(userId: number, teamId: number): Observable<any> {
    return this.http.post(`${this.baseUserUrl}${userId}/leave-team/${teamId}`, {
      responseType: 'json',
    });
  }
}
